Remove slave

lindsayad opened the pull request:
As expected, a blind master -> primary no worky. So gonna have to think about what pattern matching I can do there...

@bwspenc @dschwen @sapitts

lindsayad committed:
slave -> secondary (#15457)

lindsayad committed:
master -> primary (#15457)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/548387/) on 70e5917 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit c872d09b9ff18776002fcd0513ee5beae8c3e1ae
+Subproject commit e5d4f1aa45a4e898dffa5b0237b3210a1b15f47b
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/548566/) on 156b3a2 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/548566/) on 156b3a2 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

lindsayad committed:
Regold interface tests with slave variable names (#15457)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/548649/) on ce3deed wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/548678/) on a1b29a1 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/548678/) on a1b29a1 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/548707/) on b70d4ad wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/548707/) on b70d4ad wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

lindsayad committed:
Clang-format (#15457)

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/548762/) on 2cefe8b wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/15473/site)

This comment will be updated on new commits.

lindsayad committed:
Backwards compatability with BISON (#15457)

lindsayad commented:
@aeslaughter I have zero clues about this application documentation failure. It's complaining about not being able to locate `troubleshooting.md` but `git ls-files` shows it right in modules/doc/content/help/troubleshooting.md. Moreover I haven't touched that file nor have I touched the file that is referencing it.

aeslaughter commented:
You are not the first to see this error. Something else must of changed. I will look into it.

lindsayad committed:
Backwards compatability with Pronghorn/THM (#15457)

lindsayad committed:
Rename slave files (#15457)

lindsayad committed:
Slave -> Secondary (#15457)

dschwen commented on framework/doc/content/source/problems/DumpObjectsProblem.md:
```diff
@@ -12,7 +12,7 @@ the screen and Moose will halt execution.
 ## Example
 
-The input file `two_block_new.i` is a test for the TensorMechanics master action, an action 
+The input file `two_block_new.i` is a test for the TensorMechanics primary action, an action 
```
Ugh, that "Master" action is actually my baby...

lindsayad replied:
So we have some decisions to make. I decided against removing the [mM]aster language from multiapps because *I think* (it's been a hazy last 24 hours) it was going to required me to regold a fair amount of stuff (and now I'm wondering if it would have just been `git mv'ing` which would have been ok). And within the last 30 minutes I did a massive emacs query tags replace of Master, and chose *not* to rename the Master actions. Probably because I'm getting tired and I'm not sure how I would deprecate a Syntax gracefully (maybe it exists; I haven't researched it).

Of course if I don't rename the Master actions, I also need to go back and undo some blind "master -> primary" replacements.

I dunno man this is tough. My feeling at this point is that master in a vacuum is ok, and I'm also ok with master/sub(ordinate).

The slave replacements were easy, but the master replacements are hard and time consuming. I'm losing steam

dschwen replied:
Yeah, I feel you. I think the Master action is by far not as problematic as the master in the master/slave relationship. I suggest focussing on the contact alone for now.

lindsayad committed:
Surgical Master -> Primary (#15457)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/549668/) on 1a9712e wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

lindsayad committed:
More clang-format

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/549703/) on ac3da45 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b8ee376e5d763ff8d294333cafc7892b9f9e4706`

lindsayad commented:
Ok if all that fails is Zapdos, Griffin, and SAM (and then their dependencies) then I think this is ready to review/merge. Those applications use data (not functions) which is much harder to deprecate. We would essentially have to double our ops in the code to make sure the preferred and deprecated data are both done right.

lindsayad commented:
I have an MR submitted for Griffin. Let's see what they say over there before I remove the draft designation here

dschwen commented on framework/include/controls/ControllableItem.h:
```diff
@@ -18,8 +18,8 @@
- * The name supplied to the constructor is considered the "master" parameter. The parameter(s) added
- * via the connect method are considered the slaves.
+ * The name supplied to the constructor is considered the "master" parameter. The parameter(s)
+ * added via the connect method are considered the secondarys.
```
```suggestion
 * added via the connect method are considered the secondaries.
```

dschwen commented on framework/include/utils/InputParameters.h:
```diff
@@ -307,6 +307,18 @@ class InputParameters : public Parameters
    */
   void addCoupledVar(const std::string & name, const std::string & doc_string);
 
+  /**
+   * This method adds a deprecated coupled variable name pair.  The parser will look for variable
+   * name pair in the input file and can return a reference to the storage location
+   * for the coupled variable if found
+   * @param name The name of the coupled variable
+   * @param doc_string A description of what the coupled variable should be
+   * @param deprecation_message The deprecation message to convey to the user
+   */
+  void addDeprecatedCoupledVar(const std::string & name,
+                               const std::string & doc_string,
+                               const std::string & deprecation_message);
+
```
It'd be a better interface to declare both name here and have only one `getParam` for the new name below that does the `isParamValid` stuff behind the scenes.

lindsayad replied:
How is that going to snake its way through the Coupleable interface?

lindsayad replied:
@dschwen let me know your thoughts on this. Otherwise this is ready for more review

dschwen replied:
You need a lookup map from deprecated to non-deprecated name. That could happen in `Coupleable::getVarHelper`.

dschwen replied:
You need a lookup map from deprecated to non-deprecated name. That could happen in `Coupleable::getVarHelper`.

dschwen replied:
Coupleable has a very confusing nomenclature for what the "variable name" is (rather than the "input parameter name")...

dschwen replied:
Anyhow, this is not really central to this PR, but such a capability would make the follow up PR (and this PR) a bit simpler. When cleaning up, only the `addDeprecatedCoupledVar` lines would have to be removed.

```
   addDeprecatedCoupledVar(old_name, new_name);
```

which would make it so that the user could still provide `old_name` (but get a deprecation message). The code itself would already be updated to use `coupledValue(new_name)` (which would internally check if a deprecation mapping exists and then look for `old_name` if `new_name` hasn't been supplied - and then printing the deprecation message - which would be automatically generated from the parameter names).

lindsayad committed:
'ys' -> 'ies'

lindsayad committed:
Update for Antonios mortar PR

lindsayad committed:
_master_node_vector -> _primary_node_vector

This change needs to happen now or else there is too much inconsistency
in my patch to SAM. This will necessitate a patch for THM

lindsayad commented:
Sounds like @zachmprince is ok with my changes over in Griffin. I also have patches ready for Zapdos/SAM/THM

lindsayad commented:
@aeslaughter per the current application documentation failure [here](https://civet.inl.gov/job/550920/), should I add the `optional=True` thing like you did in #15477? What is the cause of these failures? And why don't they fail on other PR testing or next branch testing?

aeslaughter commented:
This is unrelated. There is change sitting in BISON that would fix this, but BISON isn't merging because of an HPC test that isn't running. I asked a question in slack civet-conda, but nobody cares.

lindsayad commented:
I care Andrew :-)

On Mon, Jun 22, 2020 at 9:10 PM Andrew E Slaughter <notifications@github.com>
wrote:

> This is unrelated. There is change sitting in BISON that would fix this,
> but BISON isn't merging because of an HPC test that isn't running. I asked
> a question in slack civet-conda, but nobody cares.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/15473#issuecomment-647895456>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOGA4GCRTC3SH7JX4EAMCLRYATL3ANCNFSM4OBA46CA>
> .
>

gambka commented:
> This is unrelated. There is change sitting in BISON that would fix this, but BISON isn't merging because of an HPC test that isn't running. I asked a question in slack civet-conda, but nobody cares.

If I recall correctly HPC testing targets are done on Falcon and currently that machine is down for move to C3. It can't execute if it can't start.

dschwen commented on framework/src/constraints/MortarSegmentInfo.C:
```diff
@@ -119,12 +119,12 @@ MortarSegmentInfo::is_valid() const
 
 bool
-MortarSegmentInfo::has_master() const
+MortarSegmentInfo::has_primary() const
```
I guess this should be `hasPrimary()`

lindsayad replied:
remnants of mortar's libmesh origins :-)

dschwen commented on framework/src/interfacekernels/InterfaceKernelBase.C:
```diff
@@ -59,13 +59,13 @@ InterfaceKernelBase::validParams()
       "This parameter must exist if save_in variables are specified and must have the same length "
       "as save_in. This vector specifies whether the corresponding aux_var should save-in "
-      "residual contributions from the master ('m') or slave side ('s').");
+      "residual contributions from the primary ('m') or secondary side ('s').");
```
`primary ('p')` maybe?

dschwen commented on framework/src/geomsearch/NearestNodeLocator.C:
```diff
@@ -36,7 +36,7 @@ NearestNodeLocator::NearestNodeLocator(SubProblem & subproblem,
     _subproblem(subproblem),
     _mesh(mesh),
-    _slave_node_range(NULL),
+    _secondary_node_range(NULL),
```
(could be updated to `nullptr`)

dschwen commented on modules/contact/src/constraints/MechanicalContactConstraint.C:
```diff
@@ -34,8 +34,13 @@ MechanicalContactConstraint::validParams()
+  params.addDeprecatedParam<BoundaryName>(
+      "slave",
+      "The secondary boundary",
+      "The 'slave' parameter is deprecated and will be removed on July 1, 2020.. Please use the "
```
`..`

dschwen commented on modules/doc/content/application_usage/restart_recover.md:
```diff
@@ -105,4 +105,4 @@ When running a multiapp simulation you do +not+ need to enable checkpoint output
 the default will be to continue from the previous simulation. If you supply the parameter in your input file, the new simulation will begin from
-the supplied time.
\ No newline at end of file
+the supplied time.
```
? what happened here?

dschwen commented on modules/heat_conduction/src/bcs/GapHeatTransfer.C:
```diff
@@ -356,9 +359,9 @@ GapHeatTransfer::computeQpOffDiagJacobian(unsigned jvar)
+    // FROM THE PRIMARY SURFACE WAS COMPUTED FOR A POINT ASSOCIATED
+    // WITH A SECONDARY NODE.  NOW WE ARE AT A SECONDARY INTEGRATION POINT.
     //
     // HOW DO WE GET THE NORMAL WE NEED?
```
I DON'T KNOW!! AD FOR GEOMETRIC OBJECTS MAYBE?

lindsayad replied:
![image](https://user-images.githubusercontent.com/10248304/85640579-0585b680-b641-11ea-8622-52dbfbb67785.png)

dschwen commented on modules/phase_field/include/postprocessors/FeatureFloodCount.h:
```diff
@@ -460,7 +460,7 @@ class FeatureFloodCount : public GeneralPostprocessor,
 
   /**
-   * This routine is called on the master rank only and stitches together the partial
+   * This routine is called on the primary rank only and stitches together the partial
```
not quite the same context at the contact master/slave issue, but I'd be fine with this change

dschwen reviewed (dismissed):
This should get merged sooner than later to avoid merge conflict hell for Alex. My comments are minor and could be ignored / addressed later.

bwspenc reviewed (dismissed):
Looks good to me.  My only comment is that there's no way we're going  to be able to get rid of all of the usage of the deprecated parameters by July 1. This is used in a _lot_ of input files. Maybe August 1.

lindsayad commented:
I'll address @dschwen 's comments, and work on the proposed `addDeprecatedCoupledVar` API. I'll always extend the deprecation date

lindsayad committed:
Address Daniel review

lindsayad committed:
Extend deprecation date

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/552650/) on 7e0a847 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 23da71d2668c49a62ff8081316d6496a9e0728ff`

lindsayad committed:
Add API for deprecating coupled variables (Closes #15497)

lindsayad commented:
If this passes tests (always a big if), then I think it's ready. @dschwen you can look at the last commit to see whether the `addDeprecatedCoupledVar` API looks better to you now.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/552744/) on bcb82b7 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15473/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15473/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 23da71d2668c49a62ff8081316d6496a9e0728ff`

lindsayad commented:
I guess I can't be mad if I merged the thing that creates the conflict :laughing:

dschwen commented on test/src/kernels/DeprecatedCoupledVarKernel.C:
```diff
@@ -0,0 +1,35 @@
+  params.addDeprecatedCoupledVar("stupid_name",
+                                 "source",
+                                 "A variable representing a source term",
+                                 "'stupid_name' is deprecated. Please use 'source' instead");
```
This is fine, but I'd think that the last two arguments will just be boilerplate. They could be automatically generated. But if you think the additional flexibility is necessary I'm ok with that, too.

lindsayad replied:
This reviewer drives a hard bargain! I like your idea though. I'll make the change

lindsayad committed:
Generate deprecation date and doc string for the deprecated var

Refs #15497

dschwen commented:
There is a Pronghorn failure now. Can we ignore this and fix PH later?

lindsayad commented:
Pronghorn has always failed on this PR because I have to patch THM. I have that patch ready

lindsayad commented:
@dschwen Do you want to look at the most recent change to the `addDeprecatedCoupledVar` API and potentially re-approve this? We're getting close to merging and I'm scared

dschwen commented:
Go for it. It looks like the application breakages should be easy to fix. Thanks for working on this!

lindsayad commented:
Oh boy, here comes some hell!
